mantidproject/mantid
# Indirect - Use D-Range lists across multiple runs

When the default d-range is overridden in the OSIRISDiffractionReduction algorithm, it had not impact on the used d-range, which continued to be the default. The d-range was also supplied as a single number, when it should be supplied as a list of numbers (respective to each supplied run).

The d-range is now correctly used, and is now supplied as a list to OSIRISDiffractionReduction. The interface has been updated to display this change.

This pull request also involves major refactoring of the OSIRISDiffractionReduction algorithm

**To test:**
1. Navigate to `Interfaces > Indirect > Diffraction`.
2. Ensure `OSIRIS` is selected in the Instrument drop-down menu.
3. Ensure `diffonly` is selected in the Reflection drop-down menu.
4. Enter `119977` into the run number field.
5. Ensure the Use Container checkbox is checked and enter `71987` into the field.
6. Supply a calibration file in the cal file field (calibration file included below).
7. Enter `119963` into the vanadium runs field.
8. Ensure the Manual D-Range checkbox is checked and use text field to enter a number.
9. Run the algorithm and check the *_dRange file, ensuring the entered d-range was used.
10. Repeat steps 1-8, but for step 8, try entering:
- Incorrectly formatted list string (correct format is comma separated list of numbers and number ranges written as 'a-b' where a and b are numbers). The expected result should be an error with message "Provided list, 'ENTERED TEXT', was incorrectly formatted."
- Out of bounds D-Range (e.g. 15). The expected result should be "Supplied d-range, ENTERED RANGE, is out of bounds."


Fixes #18858. 

**Release Notes** 

Added change to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
